pkg/kubelet/status/status_manager.go

```diff
@@ -415,10 +415,10 @@ func (m *manager) updateStatusInternal(pod *v1.Pod, status v1.PodStatus, forceUp
 		podName:      pod.Name,
 		podNamespace: pod.Namespace,
 	}
-	m.podStatuses[pod.UID] = newStatus
```

Comment:
See the comment under the default path, line 426:

> Let the periodic syncBatch handle the update if the channel is full.

The periodic syncBatch function grabs the status from the `podStatuses` map to do the update, but with this change the statuses map won't have the updated status, so the update will be lost.

Reply:
Suppose the channel is full, there come two versionedPodStatus's - status1 and status2.
Wouldn't the status1 be lost using the current formation (before the next syncBatch runs) ?

Reply:
It seems changing the type of podStatuses to:
```
	podStatuses      map[types.UID][]versionedPodStatus
```
would eliminate the status loss scenario described above.

Reply:
I don't think so? Most of the logic above is about reconciling the new status with the cached one. So, in your example, status2 would be merged with status1, overwriting it where relevant. We don't care that status1 is lost, just the final state is correct.

It's been a while since I thought about this though...